dace/transformation/dataflow/redundant_array.py

```diff
@@ -1326,21 +1326,22 @@ def gnode(nname):
                 # A -- (0, a:b)/(c:c+b) --> V -- (c+d)/None --> others
                 # A -- (0, a+d)/None --> others
                 e3.data.data = in_array.data
+                e3.data._is_data_src = True
```

Comment:
I see now; so it is indeed an issue with whether we can really tell which is the data source.

Reply:
Btw, this formula might be also needed in the other transformations too. I would put a note for now.

Reply:
when we add the edge it should be set automatically. in self-copies it was ambiguous though and it reset it. This line may not be necessary